src/action/common/configure_shell_profile.rs

```diff
@@ -280,27 +277,26 @@ impl Action for ConfigureShellProfile {
                     self.create_or_insert_into_files[idx] = create_or_insert_into_file
                 },
                 Ok(Err(e)) => errors.push(e),
-                Err(e) => return Err(e)?,
+                // This is quite rare and generally a very bad sign.
+                Err(e) => return Err(e).map_err(|e| Self::error(ActionErrorKind::from(e)))?,
             };
         }
 
         for create_directory in self.create_directories.iter_mut() {
-            create_directory
-                .try_revert()
-                .await
-                .map_err(|e| ActionError::Child(create_directory.action_tag(), Box::new(e)))?;
-        }
-
-        if !errors.is_empty() {
-            if errors.len() == 1 {
-                return Err(errors.into_iter().next().unwrap())?;
-            } else {
-                return Err(ActionError::Children(
-                    errors.into_iter().map(|v| Box::new(v)).collect(),
-                ));
+            if let Err(err) = create_directory.try_revert().await {
+                errors.push(err);
             }
         }
 
-        Ok(())
+        if errors.is_empty() {
```

Comment:
This pattern appears in a few places and downgrades a `MultipleErrors` to a single which allows Rust to do more error magic to it.